x-pack/plugins/security_solution/cypress/integration/overview/cti_link_panel.spec.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  OVERVIEW_CTI_ENABLE_MODULE_BUTTON,
+  OVERVIEW_CTI_LINKS,
+  OVERVIEW_CTI_LINKS_ERROR_INNER_PANEL,
+  OVERVIEW_CTI_LINKS_INFO_INNER_PANEL,
+  OVERVIEW_CTI_LINKS_WARNING_INNER_PANEL,
+  OVERVIEW_CTI_TOTAL_EVENT_COUNT,
+  OVERVIEW_CTI_VIEW_DASHBOARD_BUTTON,
+} from '../../screens/overview';
+
+import { loginAndWaitForPage } from '../../tasks/login';
+import { OVERVIEW_URL } from '../../urls/navigation';
+import { cleanKibana } from '../../tasks/common';
+import { esArchiverLoad, esArchiverUnload } from '../../tasks/es_archiver';
+
+describe('CTI Link Panel', () => {
+  before(() => {
+    cleanKibana();
+  });
+
+  describe('disabled threat intel module', () => {
+    it('renders error inner panel', () => {
+      loginAndWaitForPage(OVERVIEW_URL);
+      cy.get(`${OVERVIEW_CTI_LINKS} ${OVERVIEW_CTI_LINKS_ERROR_INNER_PANEL}`).should('exist');
+    });
+
+    it('renders disabled "view dashboard" button', () => {
+      cy.get(`${OVERVIEW_CTI_VIEW_DASHBOARD_BUTTON}`).should('be.disabled');
+    });
+
+    it('renders 0 total event count', () => {
+      cy.get(`${OVERVIEW_CTI_TOTAL_EVENT_COUNT}`).should('have.text', 'Showing: 0 events');
+    });
+
+    it('renders enable module button', () => {
+      it('renders disabled "view dashboard" button', () => {
+        cy.get(`${OVERVIEW_CTI_ENABLE_MODULE_BUTTON}`).should('exist');
+        // TODO: how to test injected doclink?
+        // cy.get(`${OVERVIEW_CTI_ENABLE_MODULE_BUTTON} a`).should('have.attr', 'href', "doclink");
+      });
+    });
+  });
+
+  describe('enabled threat intel module', () => {
+    before(() => {
+      esArchiverLoad('threat_indicator');
+    });
+
+    after(() => {
+      esArchiverUnload('threat_indicator');
+    });
+
+    describe('disabled dashboard module', () => {
+      describe('when there are no events in the selected time period', () => {
+        before(() => {
+          loginAndWaitForPage(
+            `${OVERVIEW_URL}?sourcerer=(timerange:(from:%272021-07-08T04:00:00.000Z%27,kind:absolute,to:%272021-07-09T03:59:59.999Z%27))`
+          );
+        });
+
+        it('renders warning inner panel', () => {
```

Comment:
I think cypress recommends grouping multiple assertions on the same scenario into a single test: https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion

Reply:
Since browser state is reset between each test and each test is going to loginAndWait, this should significantly reduce the time this suite takes.

Reply:
so in terms of loginAndWait, those are in `before` blocks and not a `beforeEach`, so the navigation actually happens only once for each describe block. In this suite `it` blocks are used mostly just for the developer experience, and doesn't really serve a purpose other than describing with a string what each assertion does. 

In the grouping of multiple scenarios example, each `it` block grabs the same element with `cypress.get`. Here different elements are grabbed in each block, so they can't be chained the same way.

I agree that we can group the contents of the it blocks for less verbosity, but we could be trading off the single responsibility principle. That being said, considering we also have unit test coverage, we could still consolidate. Let me know if you are happy with the new changes

Reply:
> I agree that we can group the contents of the it blocks for less verbosity, but we could be trading off the single responsibility principle.

Yeah, this is a tricky one! Normally, when I see a test with a single assertion, I know that it's an isolated case that isn't affected by any sibling tests. Cypress throws that out the window: each test shares state with the previous and subsequent tests, via both the browser and elasticsearch.

My preference (and cypress' recommendation) is to write the tests more like how they're being executed: all the data setup, browser interactions, and assertions for a particular scenario are captured in a single test.